full-moon/src/ast/luau.rs

```diff
@@ -893,114 +893,25 @@ impl ExportedTypeFunction {
-            | Self::CaretEqual(token)
-            | Self::TwoDotsEqual(token) => token,
-        }
-    }
-
-    pub(crate) fn from_token(token: TokenReference) -> Self {
-        if token.is_symbol(Symbol::PlusEqual) {
-            Self::PlusEqual(token)
-        } else if token.is_symbol(Symbol::MinusEqual) {
-            Self::MinusEqual(token)
-        } else if token.is_symbol(Symbol::StarEqual) {
-            Self::StarEqual(token)
-        } else if token.is_symbol(Symbol::SlashEqual) {
-            Self::SlashEqual(token)
-        } else if token.is_symbol(Symbol::DoubleSlashEqual) {
-            Self::DoubleSlashEqual(token)
-        } else if token.is_symbol(Symbol::PercentEqual) {
-            Self::PercentEqual(token)
-        } else if token.is_symbol(Symbol::CaretEqual) {
-            Self::CaretEqual(token)
-        } else if token.is_symbol(Symbol::TwoDotsEqual) {
-            Self::TwoDotsEqual(token)
-        } else {
-            unreachable!("converting an unknown token into a compound operator")
-        }
-    }
-}
-
-/// A Compound Assignment statement, such as `x += 1` or `x -= 1`
-#[derive(Clone, Debug, Display, PartialEq, Node, Visit)]
-#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
-#[display("{lhs}{compound_operator}{rhs}")]
-pub struct CompoundAssignment {
-    pub(crate) lhs: Var,
-    pub(crate) compound_operator: CompoundOp,
-    pub(crate) rhs: Expression,
-}
-
-impl CompoundAssignment {
-    /// Creates a new CompoundAssignment from the left and right hand side
-    pub fn new(lhs: Var, compound_operator: CompoundOp, rhs: Expression) -> Self {
-        Self {
-            lhs,
-            compound_operator,
-            rhs,
-        }
-    }
-
-    /// The variable assigned to, the `x` part of `x += 1`
-    pub fn lhs(&self) -> &Var {
-        &self.lhs
-    }
-
-    /// The operator used, the `+=` part of `x += 1`
-    pub fn compound_operator(&self) -> &CompoundOp {
-        &self.compound_operator
-    }
-
-    /// The value being assigned, the `1` part of `x += 1`
-    pub fn rhs(&self) -> &Expression {
-        &self.rhs
-    }
-
-    /// Returns a new CompoundAssignment with the given variable being assigned to
-    pub fn with_lhs(self, lhs: Var) -> Self {
-        Self { lhs, ..self }
-    }
-
-    /// Returns a new CompoundAssignment with the given operator used
-    pub fn with_compound_operator(self, compound_operator: CompoundOp) -> Self {
-        Self {
-            compound_operator,
-            ..self
-        }
-    }
-
-    /// Returns a new CompoundAssignment with the given value being assigned
-    pub fn with_rhs(self, rhs: Expression) -> Self {
-        Self { rhs, ..self }
-    }
-}
+/// A compound assignment operator, such as `+=`, `-=`, etc.
+/// This has been moved to `compound.rs` since CfxLua makes use of it as well.
+#[cfg(not(feature = "luau"))]
+#[cfg_attr(docsrs, doc(cfg(not(feature = "luau"))))]
+#[deprecated(
+    since = "full-moon 0.18.0",
```

Comment:
We're way past this already, IMO I would ditch since entirely since it doesn't add much value